cmake/legacy.cmake

```diff
@@ -370,17 +370,23 @@ endif()
 if(APPLE)
   set(root_cocoa "-Dcocoa=ON")
   set(root_x11 OFF)
+  find_package(nlohmann_json 3.9)
```

Comment:
Why only on APPLE?
Yes, you want to fix the problem on that arch.

But if we have a good enough system package on Linux, why not use it?

Reply:
Because it is/was working on the other systems without problems. ;-}}
I did not want to introduce new problems on the other supported systems, but I am also fine to add the same logic for other systems.

Reply:
Just move the logic out of the `if(APPLE)`?

… and let the CI do its job. ;-)

Let's keep our code simple, if possible?